agenta-cli/agenta/cli/main.py

```diff
@@ -22,12 +23,41 @@ def print_version(ctx, param, value):
         package_version = "package is not installed"
     click.echo(f"Agenta CLI version: {package_version}")
     ctx.exit()
+    
 
+def check_latest_version() -> Union[str, None]:
+    import requests
+    try:
+        response = requests.get('https://pypi.org/pypi/agenta/json')
+        response.raise_for_status()
+        latest_version = response.json()['info']['version']
+        return latest_version
+    except (requests.RequestException, KeyError):
+        return None
+
+
+def notify_update(available_version: str):
+    import pkg_resources
+    installed_version = pkg_resources.get_distribution('agenta').version
+    if available_version > installed_version:
+        click.echo(
+            click.style(
+                f"A new release of agenta is available: {installed_version} → {available_version}",
+                fg="yellow",
+            )
+        )
+        click.echo(
+            click.style(
+                "To upgrade, run: pip install --upgrade agenta", fg="yellow"
+            )
+        )
```

Comment:
**Suggestion:** Consider using the 'packaging' library to compare versions. This will handle edge cases where string comparison might fail.
```suggestion
from packaging import version

def notify_update(available_version: str):
    import pkg_resources
    installed_version = pkg_resources.get_distribution('agenta').version
    if version.parse(available_version) > version.parse(installed_version):
        click.echo(
            click.style(
                f"A new release of agenta is available: {installed_version} → {available_version}",
                fg="yellow",
            )
        )
        click.echo(
            click.style(
                "To upgrade, run: pip install --upgrade agenta", fg="yellow"
            )
        )
```